src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts

```diff
@@ -608,17 +618,49 @@ export function getGeneticTracks(
         data.filter(d => !(d.sample in excludedSampleIdsMap))
     );
 
+    // note to AARON.  we need to fill out gapLels function for each track
+    // but note that the sequencing maps aren't important because
+    // in oncoprinter, we ignore when something has been sequenced.  it's data
+    // which is not pertinent in this context.
+
     const geneToPercentAltered: {
-        [hugoGeneSymbol: string]: string;
+        [hugoGeneSymbol: string]: any;
     } = _.mapValues(geneToOncoprintData, getPercentAltered);
     const genes = geneOrder
         ? geneOrder.filter(gene => gene in geneToOncoprintData)
         : Object.keys(geneToOncoprintData);
     return genes.map(gene => ({
         key: getGeneticTrackKey(gene),
         label: gene,
-        info: geneToPercentAltered[gene],
+        info: geneToPercentAltered[gene].percent,
         data: geneToOncoprintData[gene],
+        customOptions: [
+            {
+                gapLabelsFn: (model: OncoprintModel) => {
+                    model.data_groups.update(model);
+                    const groupsByTrackMap = model.data_groups.get();
+
+                    const percentagesForGroups = groupsByTrackMap[gene][0].map(
+                        groupData => {
+                            return getPercentAltered(
+                                groupData as OncoprinterGeneticTrackDatum[]
+                            );
+                        }
+                    );
```

Comment:
would be simpler if we could do something like

```suggestion
                    const percentagesForGroups = groupsByTrackMap[gene][0].map(getPercentAltered);
```